src/main/webapp/pipe.js

```diff
@@ -311,124 +283,150 @@ function pipelineUtils() {
     }
 }
 
+function addPipelineHeader(html, component, data, c, resURL) {
+    html.push('<h1>' + htmlEncode(component.name));
+    if (data.allowPipelineStart) {
+        if (component.workflowComponent) {
+            html.push('&nbsp;<a id="startpipeline-' + c  +'" class="task-icon-link" href="#" onclick="triggerBuild(\'' + component.workflowUrl + '\', \'' + data.name + '\')">');
+        } else if (component.firstJobParameterized) {
+            html.push('&nbsp;<a id="startpipeline-' + c  +'" class="task-icon-link" href="#" onclick="triggerParameterizedBuild(\'' + component.firstJobUrl + '\', \'' + data.name + '\')">');
+        } else {
+            html.push('&nbsp;<a id="startpipeline-' + c  +'" class="task-icon-link" href="#" onclick="triggerBuild(\'' + component.firstJobUrl + '\', \'' + data.name + '\')">');
+        }
+        html.push('<img class="icon-clock icon-md" title="Build now" src="' + resURL + '/images/24x24/clock.png">');
+        html.push('</a>');
+    }
+    html.push('</h1>');
+}
+
+function checkDataForError(data, errrorDivId) {
+    var cErrorDiv = Q('#' + errrorDivId);
+    if (data.error) {
+        cErrorDiv.html('Error: ' + data.error).show();
+    } else {
+        cErrorDiv.hide().html('');
+    }
+}
+
+function isTaskLoggedToConsole(task) {
```

Comment:
Passing data as parameter is fixed in the next commit of this pull request. About naming: I would like to leave functions/methods returning boolean value names starting with "is", because such naming tells that method returning "is truthy" or "is falsy". When method is called "add...", it usually means that method will add something to some data model. For example, addNewItem(destArray) and isAddingAllowed(destArray) describes the idea.

What do you think about such approach?